Add UI for editing attachment rename format pref

tnajdek opened the pull request:
This PR adds input box to edit `extensions.zotero.attachmentRenameFormatString` preference, including a preview for how a currently selected item would be used to derive a file name. If no item is selected, preview will use a mocked item.

`helpURL` is set to a [section](https://www.zotero.org/support/preferences/general#file_renaming) on a support page that does not yet exist. I'm happy to write a couple of paragraphs explaining how this works.

![Screenshot 2023-07-25 at 18 49 02](https://github.com/zotero/zotero/assets/214628/b6f41c0f-820c-4d36-b69d-7962df9514f7)

dstillman commented:
I think in this case the help button is too subtle and we need an actual line of text with a link.

tnajdek commented:
* Updated to use the new `extensions.zotero.attachmentRenameTemplate` preference
* Removed help button and instead added brief instructions and a text link

![Screenshot 2023-08-04 at 18 53 06](https://github.com/zotero/zotero/assets/214628/14c72f03-5022-46ec-96f5-884a8a26c164)

dstillman commented:
I've started a dedicated page for this:

https://www.zotero.org/support/file_renaming

I can add more details as necessary, but it's probably easier if you fill in the variables and parameters at the end. And I guess maybe we need some examples for conditionals, like on the [note templates](https://www.zotero.org/support/note_templates) page.

tnajdek commented:
Thanks! I'll edit the page and update the link in this PR.

dstillman commented:
Merged in d32b061 and 4e3977e
